File: database/src/main/java/hr/foi/air/database/entities/Setting.java
package hr.foi.air.database.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "settings")
public class Setting {
    @PrimaryKey(autoGenerate = true)
    int id;
    @ForeignKey(entity = User.class, parentColumns = "id", childColumns = "user_id")
    @ColumnInfo(index = true, name = "user_id")
    int userId;

    String key;
    String value;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: server/database/entities/Setting.php
<?php

class Setting {
    private $id;
    private $key;
    private $value;
    private $userId;
    
    public function getId() {
        return $this->id;
    }

    public function getKey() {
        return $this->key;
    }

    public function getValue() {
        return $this->value;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function setId($id): void {
        $this->id = $id;
    }

    public function setKey($key): void {
        $this->key = $key;
    }

    public function setValue($value): void {
        $this->value = $value;
    }

    public function setUserId($userId): void {
        $this->userId = $userId;
    }
    
    public static function fromObject($obj) {
        $setting = new Setting();
        if (isset($obj->id)) {
            $setting->id = $obj->id;
        }
        if (isset($obj->key)) {
            $setting->key = $obj->key;
        }
        if (isset($obj->value)) {
            $setting->value = $obj->value;
        }
        if (isset($obj->userId)) {
            $setting->userId = $obj->userId;
        }
        return $setting;
    }

}